deploy/v3_migration_operator.js

```diff
@@ -0,0 +1,53 @@
+const { ethers } = require('hardhat');
+
+/**
+ * Deploys the Migration Operator contract to migrate the controller & terminal.
+ *
+ * Example usage:
+ *
+ * npx hardhat deploy --network goerli --tags JB_Migration_Operator
+ */
+module.exports = async ({ deployments, getChainId }) => {
+  console.log('Deploying v3 Migration Operator');
+
+  const { deploy } = deployments;
+  const [deployer] = await ethers.getSigners();
+
+  let multisigAddress;
+  let jbDirectory;
+  let chainId = await getChainId();
+  let baseDeployArgs = {
+    from: deployer.address,
+    log: true,
+    skipIfAlreadyDeployed: true,
+  };
+
+  console.log({ deployer: deployer.address, chain: chainId });
+
+  switch (chainId) {
+    // mainnet
+    case '1':
+      multisigAddress = '0xAF28bcB48C40dBC86f52D459A6562F658fc94B1e';
+      jbDirectory = '0x65572FB928b46f9aDB7cfe5A4c41226F636161ea';
```

Comment:
You shouldn't have to hardcode the addresses, hardhat deploy will check the previous deployment (and skip redeploying as you pass the arg), more future proof then (v8.3 or smth;)

Reply:
but we'll need the address right either by importing the deployment json Or hardcoding Or maybe I misunderstood something?

Reply:
hardhat-deploy check the deployments metadata in deploy and (when using the arg) doesn't redeploy/reuse contracts already deployed (if you run 1.js for instance, you'll get a series of "Already deployed-reusing")

Reply:
Oh weird didn't know it behaves like this even when doing a force deployment, I'll then just import the address from the json

Reply:
wait, no @viraj124 , I'll suggest change what I meant;)